hack/docs/instancetypes_gen_docs.go

```diff
@@ -195,3 +199,47 @@ below are the resources available with some assumptions and after the instance o
 		}
 	}
 }
+
+type kubeDnsTransport struct {
+}
+
+const kubeDNS = `{
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {
+        "creationTimestamp": "2022-04-14T17:55:49Z",
+        "name": "kube-dns",
+        "namespace": "kube-system",
+        "resourceVersion": "262"
+    },
+    "spec": {
+        "clusterIP": "10.100.0.10",
+        "clusterIPs": [
+            "10.100.0.10"
+        ],
+        "internalTrafficPolicy": "Cluster",
+        "ipFamilies": [
+            "IPv4"
+        ],
+        "ipFamilyPolicy": "SingleStack",
+        "type": "ClusterIP"
+    }
+}
+`
+
+func (f kubeDnsTransport) RoundTrip(request *http.Request) (*http.Response, error) {
+	return &http.Response{
+		StatusCode: http.StatusOK,
+		Proto:      "http/1.0",
+		ProtoMajor: 1,
+		ProtoMinor: 0,
+		Body:       io.NopCloser(bytes.NewBufferString(kubeDNS)),
+	}, nil
+}
+
+func NewAWSCloudProviderForCodeGen() *awscloudprovider.CloudProvider {
+	cs, _ := kubernetes.NewForConfigAndClient(&rest.Config{}, &http.Client{Transport: &kubeDnsTransport{}})
```

Comment:
This requires you to have access to an authenticated cluster to generate the docs right? How does this work in the codegen?